git_machete/cmd.py

```diff
@@ -2263,95 +1662,34 @@ def contains_equivalent_tree(
     return result
 
 
-def create_branch(cli_ctxt: CommandLineContext, b: str, out_of_revision: str) -> None:
-    run_git(cli_ctxt, "checkout", "-b", b, out_of_revision)
+def create_branch(cli_ctxt: CommandLineContext, b: str, out_of_revision: str) -> None:  # maybe move such a methods into github.py
+    git_operations.run_git(cli_ctxt, "checkout", "-b", b, out_of_revision)
     flush_caches()  # the repository state has changed b/c of a successful branch creation, let's defensively flush all the caches
 
 
 def log_shas(cli_ctxt: CommandLineContext, revision: str, max_count: Optional[int]) -> List[str]:
     opts = ([f"--max-count={str(max_count)}"] if max_count else []) + ["--format=%H", f"refs/heads/{revision}"]
-    return non_empty_lines(popen_git(cli_ctxt, "log", *opts))
+    return utils.non_empty_lines(git_operations.popen_git(cli_ctxt, "log", *opts))
 
 
 MAX_COUNT_FOR_INITIAL_LOG = 10
 
-initial_log_shas_cached: Dict[str, List[str]] = {}
-remaining_log_shas_cached: Dict[str, List[str]] = {}
-
 
 # Since getting the full history of a branch can be an expensive operation for large repositories (compared to all other underlying git operations),
 # there's a simple optimization in place: we first fetch only a couple of first commits in the history,
 # and only fetch the rest if none of them occurs on reflog of any other branch.
 def spoonfeed_log_shas(cli_ctxt: CommandLineContext, b: str) -> Generator[str, None, None]:
-    if b not in initial_log_shas_cached:
-        initial_log_shas_cached[b] = log_shas(cli_ctxt, b, max_count=MAX_COUNT_FOR_INITIAL_LOG)
-    for sha in initial_log_shas_cached[b]:
+    if b not in GitContext.initial_log_shas_cached:
```

Comment:
Hmm I think this method is Machete-agnostic enough that it can go directly to `GitContext`